src/Stratis.Bitcoin.Features.PoA/CollateralPoAConsensusFactory.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Text;
+using NBitcoin;
+using Stratis.Bitcoin.Utilities.JsonConverters;
+
+namespace Stratis.Bitcoin.Features.PoA
+{
+    public class CollateralPoAConsensusFactory : PoAConsensusFactory
+    {
+        public override IFederationMember DeserializeFederationMember(byte[] serializedBytes)
+        {
+            string json = Encoding.ASCII.GetString(serializedBytes);
+
+            CollateralFederationMemberModel model = Serializer.ToObject<CollateralFederationMemberModel>(json);
+
+            var member = new CollateralFederationMember(new PubKey(model.PubKeyHex), new Money(model.CollateralAmountSatoshis), model.CollateralMainchainAddress);
+
+            return member;
+        }
+
+        public override byte[] SerializeFederationMember(IFederationMember federationMember)
+        {
+            var member = federationMember as CollateralFederationMember;
+
+            if (member == null)
+                throw new ArgumentException($"Member of type: '{nameof(CollateralFederationMember)}' should be provided.");
+
+            var model = new CollateralFederationMemberModel()
+            {
+                CollateralMainchainAddress = member.CollateralMainchainAddress,
```

Comment:
Can any of these be null? If not rather create the object as a value object? I.e. pass all the values into the constructor.

Reply:
>Can any of these be null?

Yes